### ISSUE TITLE
The IDE should never throw an exception

### ISSUE THREAD

**Issue by rickmcgeer** [LivelyKernel/lively.next#1384]
**Describe the Feature**
I almost filed this as a bug report.  The issue is that when there's a bug in the user's code that causes an IDE exception, the error messages are the last word in unhelpful.  In the first place, the line references are never to the user's code, and for that matter aren't even line references to the source code in the lively distribution (there would still be issues if they were, because the editors don't show line numbers); they're references to a line in the lively build, which can only be accessed through the browser debugger.  If the programmer navigates to that line, he _might_ recognize the line in the original code, and from there take a stab at debugging it.

I spent today finding that _one line of configuration_ of a component wasn't supported.  Specifically, it was the line `renderGPU: true`
in 
````
$world.execCommand("open browser", {moduleName: "studio/inputs/search.cp.js", packageName: "engageLively--galyleo-dashboard", codeEntity: "dropShadow"});
```

And it was a detective hunt, because the IDE gave me no help at all.  I started off trying to instantiate `BugReporter` in

```
$world.execCommand("open browser", {moduleName: "studio/helpers.cp.js", packageName: "engageLively--galyleo-dashboard", codeEntity: "BugReporter"});
```


which  failed, tossing a red-box error message that was unhelpful.  Hitting "Edit Component" did nothing.  Figuring that it had to be one of the submorphs, I tried instantiating each in turn until I hit `GalyleoSearch`, which crapped out.  I then tried instantiating new components which were clones until it stopped working, which is where I found the fatal line.  I commented it out and stuff worked all up the chain. 

The point is, for most people this is a walkaway.  I love this environment, so I don't give up, but I know this needs to be fixed.  And I know it's a big job.

Ultimately, the issue is defensive code in the IDE -- if a user's code is going to cause the IDE to hit an exception, that IDE needs to return a meaningful error message with the chunk of the user's code that's causing the problem.  I appreciate this is a big job, but I don't see an alternative...

**Graphics**
If applicable, provide screenshots/videos of similar ideas in other software.

**Implementation Details**
If applicable, provide insights with regards to implementing this feature that you already have.

**Comment by merryman** [comment]
From what I can tell this would be alleviated to some degree once we support source maps #316 and enhancing the error messages to allow users to jump to each affected line.

**Comment by linusha** [comment]
I agree with @merryman regarding the source-maps being helpful in such cases.

**@rickmcgeer in this concrete scenario however, you did nothing wrong!** Your component definition is absolutely fine, but you hit a bug in the rendering logic of the system. I will open a PR that fixes that, then you can use `renderOnGPU` just fine like you should be able to. 
I would then close this ticket, as we have a separate one for the source-maps and we actually cannot do anything against cases like this. The solution to such problems is basically to have the lively core bug free, which we of course strive to achieve, but sometimes all of us create little :bug:...
Just to reiterate: The problem here is not with your code, but you uncovered a bug in the system, so this is indeed more of a bug report. The render bug will be fixed soon :slightly_smiling_face:
